### PR TITLE
Add monitoring to GCP

### PR DESCRIPTION
**Pull request by juadk**
Enable monitoring for GCP with an option to disable it

GCP documentation needs to be rework as is in AWS or Azure (future PR).

### CODE REVIEW

File: gcp/outputs.tf [hunk 1]
```diff
@@ -1,10 +1,56 @@
...
+output "cluster_nodes_public_name" {
+  value = []
+}
+
+# Monitoring
+
+output "monitoring_nodes_ip" {
+  value = google_compute_instance.monitoring.*.network_interface.0.network_ip
+}
+output "monitoring_nodes_public_ip" {
```
Comment (ldevulder) [inline gcp/outputs.tf]: An empty line could help the reading ;-)

File: gcp/outputs.tf [hunk 2]
```diff
@@ -1,10 +1,56 @@
...
+  value = google_compute_instance.monitoring.*.network_interface.0.access_config.0.nat_ip
+}
+output "monitoring_nodes_name" {
+  value = ""
+}
```
Comment (ldevulder) [inline gcp/outputs.tf]: Should be: `value = google_compute_instance.monitoring.*.name`

File: gcp/disks.tf [hunk 3]
```diff
@@ -21,3 +21,9 @@ resource "google_compute_disk" "node_data2" {
   zone  = element(data.google_compute_zones.available.names, count.index)
 }
 
+resource "google_compute_disk" "monitoring_data" {
+  name = "${terraform.workspace}-${var.name}-monitoring-data"
```
Comment (arbulu89) [inline gcp/disks.tf]: I guess we don't need to create the disk if the monitoring is not enabled, right?
We could add the same condition than in the VM

Comment (juadk) [reply gcp/disks.tf]: > I guess we don't need to create the disk if the monitoring is not enabled, right?
> We could add the same condition than in the VM

You are right, good catch thx

Comment (juadk) [reply gcp/disks.tf]: @arbulu89 Done

**Comment by juadk** [comment]
> Without testing personally the changes it looks good and ready to be merged! Thanks @juadk

Thanks @arbulu89 , I've tested it quickly with and without the monitoring part and it worked well.

**Pull request merged**